README.md

```diff
@@ -1,3 +1,166 @@
+
+```es6
+class Rectangle extends Component {
+  static propTypes = {
+    layout: PropTypes.shape({
+      top: PropTypes.number,
+      right: PropTypes.number,
+      bottom: PropTypes.number,
+      left: PropTypes.number
+    })
+  };
+
+  render() {
+    return (
+      <div style={{
+        position: "absolute",
+        top: this.props.layout.top || 0,
+        left: this.props.layout.left || 0
+        width: this.props.layout.width || 0,
+        height: this.props.layout.height || 0
+      }}
+      >
+        <p>It's demo time!</p>
+      </div>
+    );
+  }
+}
+
+export default Subview(Rectangle);
+```
+
+Here's how to use `AutoDOM` components:
+
+```es6
+import ConstraintLayout, { View, AutoDOM } from "radium-constraints";
+
+<ConstraintLayout>
+  <View
+    name="tutorial"
+    container="div"
+    width={400}
+    height={500}
+    style={{
+      background: "red"
+    }}
+  >
+    <AutoDOM.div
+      name="tutorialSubview"
+      intrinsicWidth={50}
+      intrinsicHeight={50}
+      constraints={[
+        constrain().subview("tutorialSubview").centerX
+          .to.equal.superview.centerX,
+        constrain().subview("tutorialSubview").centerY
+          .to.equal.superview.centerY
+      ]}
+    >
+      This is a subview
+    </AutoDOM.div>
+  </View>
+</ConstraintLayout>
+```
+
+When using `AutoSVG` components, make sure to pass "g" instead of "div" to the `<View>`'s `container` prop.
+
+## Demo
+There are more complex examples on the demo page. Check out the code in [app.jsx](https://github.com/FormidableLabs/radium-constraints/blob/master/demo/app.jsx).
+
+### Installation
+- Install builder: `npm install -g builder`
+- Clone this repo
+- `npm install` and then `builder run hot` will load a webpack dev server at localhost:3000
+
+## Caveats
+React Constraints uses an asynchronous layout engine running on a pool of WebWorkers. This prevents layout calculations from bogging down the main thread and allows the library to be a good citizen in any page/app. Therefore, browsers using this library must support WebWorkers.
+
+Resolving and incrementally adding/removing constraints are cheap enough to run in 60fps for most cases. However, the initial layout calculations on first load are the most expensive, and you may notice a slight delay in layout (although this does not block the main thread). We're working on a build tool that will pre-calculate initial layouts and feed them into your components to prevent this.
+
+## Roadmap <a id="roadmap"></a>
+In order of priority:
+- Remove dependency on autolayout.js in favor of a simple wrapper around the Kiwi constraint solver.
+- Create build tool to pre-calculate initial layouts.
+- Decide on a distribution strategy for the WebWorker code (preference on inlining).
+- Decide on an animation strategy (requires support for removing constraints).
+- Allow for self-referential subviews in the constraint props array without using the subview string.
```

Comment:
If it were just for me, I'd vote for docstrings - I like reading through the source.  But since this is a public package, I guess explicit documentation in the README would be better.